commit f942328eeb552728dab6780087e28f026cb66934
Author: Loriane Hilderal <loriane.hilderal@canal-plus.com>
Date:   Wed May 14 23:24:22 2025 +0200

    sending auth token to backend

diff --git a/src/components/AddStudentToStudentBatch/AddStudentModal.tsx b/src/components/AddStudentToStudentBatch/AddStudentModal.tsx
index 9b7895d..4d37c8f 100644
--- a/src/components/AddStudentToStudentBatch/AddStudentModal.tsx
+++ b/src/components/AddStudentToStudentBatch/AddStudentModal.tsx
@@ -26,7 +26,7 @@ export default function AddStudentModal({
     const [isLoading, setIsLoading] = useState(false)
     const [studentData, setStudentData] = useState<Student>({
         last_name: '',
-        id: 'nobody',
+        user_id: 'nobody',
         first_name: '',
         email: '',
     })
diff --git a/src/components/AddStudentToStudentBatch/DraggableStudent.tsx b/src/components/AddStudentToStudentBatch/DraggableStudent.tsx
index b5b72b1..9e9d4e0 100644
--- a/src/components/AddStudentToStudentBatch/DraggableStudent.tsx
+++ b/src/components/AddStudentToStudentBatch/DraggableStudent.tsx
@@ -3,7 +3,7 @@ import { Student } from '@/components/ManageStudentBatches/types.ts'
 
 export const DraggableStudent = ({ student }: { student: Student }) => {
     const { attributes, listeners, setNodeRef, transform } = useDraggable({
-        id: student.id,
+        id: student.user_id,
     })
 
     return (
diff --git a/src/components/AddStudentToStudentBatch/StudentBatchDroppableContainers.tsx b/src/components/AddStudentToStudentBatch/StudentBatchDroppableContainers.tsx
index 04a6b99..d937d0d 100644
--- a/src/components/AddStudentToStudentBatch/StudentBatchDroppableContainers.tsx
+++ b/src/components/AddStudentToStudentBatch/StudentBatchDroppableContainers.tsx
@@ -8,43 +8,43 @@ import AddStudentModal from '@/components/AddStudentToStudentBatch/AddStudentMod
 
 const initialStudents = [
     {
-        id: '1',
+        user_id: '1',
         first_name: 'Lara',
         last_name: 'CROFT',
         email: 'lara@gmail.com',
     },
     {
-        id: '2',
+        user_id: '2',
         first_name: 'Max',
         last_name: 'CHARLES',
         email: 'nino@gmail.com',
     },
     {
-        id: '3',
+        user_id: '3',
         first_name: 'Lara',
         last_name: 'CROFT',
         email: 'lara@gmail.com',
     },
     {
-        id: '42',
+        user_id: '42',
         first_name: 'Max',
         last_name: 'CHARLES',
         email: 'nino@gmail.com',
     },
     {
-        id: '492',
+        user_id: '492',
         first_name: 'Max',
         last_name: 'CHARLES',
         email: 'nino@gmail.com',
     },
     {
-        id: '20',
+        user_id: '20',
         first_name: 'Max',
         last_name: 'CHARLES',
         email: 'nino@gmail.com',
     },
     {
-        id: '219',
+        user_id: '219',
         first_name: 'Max',
         last_name: 'CHARLES',
         email: 'nino@gmail.com',
@@ -60,20 +60,20 @@ export default function StudentBatchDroppableContainers({
     setSelectedStudents,
 }: StudentBatchDroppableContainersProps) {
     const [allStudents, setAllStudents] = useState<Student[]>(initialStudents) //todo: getAllStudents from the DB or from the teacher Default promotion ?
-    const selectedIds = selectedStudents.map((s) => s.id)
+    const selectedIds = selectedStudents.map((s) => s.user_id)
     const availableStudents = allStudents.filter(
-        (s) => !selectedIds.includes(s.id)
+        (s) => !selectedIds.includes(s.user_id)
     )
 
     const handleDragEnd = (event: DragEndEvent) => {
         const { over, active } = event
         if (over?.id === 'selected' && active?.id) {
             const draggedStudent = availableStudents.find(
-                (s) => s.id === active.id
+                (s) => s.user_id === active.id
             )
             if (
                 draggedStudent &&
-                !selectedStudents.some((s) => s.id === active.id)
+                !selectedStudents.some((s) => s.user_id === active.id)
             ) {
                 setSelectedStudents([...selectedStudents, draggedStudent])
             }
@@ -81,12 +81,12 @@ export default function StudentBatchDroppableContainers({
     }
 
     const handleRemove = (id: string) => {
-        setSelectedStudents(selectedStudents.filter((s) => s.id !== id))
+        setSelectedStudents(selectedStudents.filter((s) => s.user_id !== id))
     }
 
     const handleAddAll = () => {
         const toAdd = availableStudents.filter(
-            (s) => !selectedStudents.find((ss) => ss.id === s.id)
+            (s) => !selectedStudents.find((ss) => ss.user_id === s.user_id)
         )
         setSelectedStudents(toAdd)
     }
@@ -114,7 +114,7 @@ export default function StudentBatchDroppableContainers({
                                 ) : (
                                     availableStudents.map((student) => (
                                         <DraggableStudent
-                                            key={student.id}
+                                            key={student.user_id}
                                             student={student}
                                         />
                                     ))
@@ -149,7 +149,7 @@ export default function StudentBatchDroppableContainers({
                                 ) : (
                                     selectedStudents.map((student) => (
                                         <div
-                                            key={student.id}
+                                            key={student.user_id}
                                             className="p-2 m-2 bg-white dark:bg-sidebar-accent border rounded shadow cursor-move flex items-center gap-2"
                                         >
                                             <span>
@@ -159,7 +159,9 @@ export default function StudentBatchDroppableContainers({
                                             </span>
                                             <button
                                                 onClick={() =>
-                                                    handleRemove(student.id)
+                                                    handleRemove(
+                                                        student.user_id
+                                                    )
                                                 }
                                                 style={{ cursor: 'pointer' }}
                                                 className="text-red-500 hover:text-red-700"
diff --git a/src/components/LogIn/LogIn.tsx b/src/components/LogIn/LogIn.tsx
index 3e5b6b0..e3199c2 100644
--- a/src/components/LogIn/LogIn.tsx
+++ b/src/components/LogIn/LogIn.tsx
@@ -34,6 +34,7 @@ export default function LogIn({
             const response: UserLoginResponse =
                 await authService.login(formData)
             if (response.success) {
+                toast.success(`Logged in successfully: ${response.token}`)
                 window.location.href = '/'
             } else {
                 toast.error(response.error)
diff --git a/src/components/LogIn/types.ts b/src/components/LogIn/types.ts
index 39a7f00..4b5a1c7 100644
--- a/src/components/LogIn/types.ts
+++ b/src/components/LogIn/types.ts
@@ -6,4 +6,5 @@ export type UserLoginDto = {
 export type UserLoginResponse = {
     success: boolean
     error?: string
+    token?: string
 }
diff --git a/src/components/ManageStudentBatches/StudentBatchById.tsx b/src/components/ManageStudentBatches/StudentBatchById.tsx
index 02dd731..404dd87 100644
--- a/src/components/ManageStudentBatches/StudentBatchById.tsx
+++ b/src/components/ManageStudentBatches/StudentBatchById.tsx
@@ -33,7 +33,6 @@ export default function StudentBatchById() {
         setBatchEditData((prev) => ({ ...prev, [id]: value }))
     }
     const updateSelectedStudents = (selectedStudents: Student[]) => {
-        console.log(selectedStudents)
         setBatchEditData((prev) => ({
             ...prev,
             students: [...selectedStudents],
@@ -132,8 +131,7 @@ export default function StudentBatchById() {
                                     onChange={handleChange}
                                 />
                             </TableCell>
-                            <TableCell>{0}</TableCell>
-                            {/*TODO: add batch.students/length later*/}
+                            <TableCell>{batch.students.length}</TableCell>
                             <TableCell>
                                 {formatToShortDate(batch.createdAt)}
                             </TableCell>
diff --git a/src/components/ManageStudentBatches/StudentBatchesPage.tsx b/src/components/ManageStudentBatches/StudentBatchesPage.tsx
index 7eaad9f..998e6c6 100644
--- a/src/components/ManageStudentBatches/StudentBatchesPage.tsx
+++ b/src/components/ManageStudentBatches/StudentBatchesPage.tsx
@@ -22,7 +22,8 @@ import { useNavigate } from 'react-router'
 import { batchService } from '@/services/UserService/auth-api-client.ts'
 import { useEffect, useState } from 'react'
 import AddStudentBatchModal from '@/components/ManageStudentBatches/AddStudentBatchModal.tsx'
-import {formatToShortDate} from "@/utils/dateFormatter.ts";
+import { formatToShortDate } from '@/utils/dateFormatter.ts'
+
 export default function StudentBatchesPage() {
     const navigate = useNavigate()
     const [isLoading, setIsLoading] = useState(false)
@@ -112,8 +113,10 @@ export default function StudentBatchesPage() {
                                 </Badge>
                             </TableCell>
                             <TableCell>{batch.name}</TableCell>
-                            <TableCell>{0}</TableCell> {/*TODO: add batch.students/length later*/}
-                            <TableCell>{formatToShortDate(batch.createdAt)}</TableCell>
+                            <TableCell>{batch.students.length}</TableCell>
+                            <TableCell>
+                                {formatToShortDate(batch.createdAt)}
+                            </TableCell>
                             <TableCell>{batch.tags}</TableCell>
                             <TableCell>
                                 <DropdownMenu>
diff --git a/src/components/ManageStudentBatches/mocks.ts b/src/components/ManageStudentBatches/mocks.ts
index 2dc2bb0..d867db6 100644
--- a/src/components/ManageStudentBatches/mocks.ts
+++ b/src/components/ManageStudentBatches/mocks.ts
@@ -9,7 +9,7 @@ export const severalStudentBatches: StudentBatch[] = [
         name: 'Default Promotion',
         students: [
             {
-                id: 'first',
+                user_id: 'first',
                 first_name: 'Loriane',
                 last_name: 'HILDERAL',
                 email: 'loriane@gmail.com',
@@ -24,13 +24,13 @@ export const severalStudentBatches: StudentBatch[] = [
         name: 'Promotion A',
         students: [
             {
-                id: 'first',
+                user_id: 'first',
                 first_name: 'Loriane',
                 last_name: 'HILDERAL',
                 email: 'loriane@gmail.com',
             },
             {
-                id: 'first',
+                user_id: 'first',
                 first_name: 'Nino',
                 last_name: 'RUTH',
                 email: 'nino@gmail.com',
@@ -47,7 +47,7 @@ export const oneStudentStudentBatch: StudentBatch = {
     name: 'Default Promotion',
     students: [
         {
-            id: 'first',
+            user_id: 'first',
             first_name: 'Loriane',
             last_name: 'HILDERAL',
             email: 'loriane@gmail.com',
diff --git a/src/components/ManageStudentBatches/types.ts b/src/components/ManageStudentBatches/types.ts
index a0d8131..79fdfb3 100644
--- a/src/components/ManageStudentBatches/types.ts
+++ b/src/components/ManageStudentBatches/types.ts
@@ -20,7 +20,7 @@ export type NewStudentBatch = {
 }
 
 export type Student = {
-    id: string
+    user_id: string
     email: string
     first_name: string
     last_name: string
diff --git a/src/services/UserService/auth-api-client.ts b/src/services/UserService/auth-api-client.ts
index 26ab0ec..ef9b4c9 100644
--- a/src/services/UserService/auth-api-client.ts
+++ b/src/services/UserService/auth-api-client.ts
@@ -37,16 +37,22 @@ export const batchService = {
     getOneById: async (id: string): Promise<BatchServiceResponse> => {
         try {
             const response = await fetch(
-                `${PROJECT_API_URL}/student-batches/${id}`
+                `${PROJECT_API_URL}/student-batches/${id}`,
+                {
+                    headers: {
+                        Authorization: `Bearer ${localStorage.getItem('auth_token')}`,
+                    },
+                }
             )
             if (!response.ok) {
                 const error: ApiErrorMessage = await response.json()
                 return handleApiError(error.message)
             } else {
                 const batch: StudentBatch = await response.json()
+                console.log(batch)
                 return {
                     success: true,
-                    data: { ...batch, students: [] }, //TODO: change when backend sends students
+                    data: batch,
                 }
             }
             // return {
@@ -60,7 +66,11 @@ export const batchService = {
     },
     getAll: async (): Promise<BatchServiceResponse> => {
         try {
-            const response = await fetch(`${PROJECT_API_URL}/student-batches`)
+            const response = await fetch(`${PROJECT_API_URL}/student-batches`, {
+                headers: {
+                    Authorization: `Bearer ${localStorage.getItem('auth_token')}`,
+                },
+            })
             if (!response.ok) {
                 const error: ApiErrorMessage = await response.json()
                 return handleApiError(error.message)
@@ -116,7 +126,10 @@ export const batchService = {
         try {
             const response = await fetch(`${PROJECT_API_URL}/student-batches`, {
                 method: 'POST',
-                headers: { 'Content-Type': 'application/json' },
+                headers: {
+                    'Content-Type': 'application/json',
+                    Authorization: `Bearer ${localStorage.getItem('auth_token')}`,
+                },
                 body: JSON.stringify(createBatchData),
             })
             if (!response.ok) {
@@ -145,6 +158,9 @@ export const batchService = {
                 `${PROJECT_API_URL}/student-batches/${id}`,
                 {
                     method: 'DELETE',
+                    headers: {
+                        Authorization: `Bearer ${localStorage.getItem('auth_token')}`,
+                    },
                 }
             )
             if (!response.ok) {
@@ -157,11 +173,11 @@ export const batchService = {
                     data: batch, //TODO: change when backend sends students
                 }
             }
-        }catch (error) {
+        } catch (error) {
             const err = error as ApiErrorMessage
             return handleApiError(err.message)
         }
-    }
+    },
 }
 export const authService = {
     register: async (
@@ -186,19 +202,28 @@ export const authService = {
             return handleApiError(err.message)
         }
     },
+    // login: async (loginData: UserLoginDto): Promise<UserLoginResponse> => {
+    //     console.log(loginData)
+    //     return { success: false, error: 'Not implemented yet' }
+    // },
     login: async (loginData: UserLoginDto): Promise<UserLoginResponse> => {
-        console.log(loginData)
-        return { success: false, error: 'Not implemented yet' }
-    },
-    ssoLogin: async () => {
         try {
             const response = await fetch(
-                `${AUTH_API_URL}/user-api/auth/callback/google`
+                `${AUTH_API_URL}/user-api/login/teacher`,
+                {
+                    method: 'POST',
+                    headers: { 'Content-Type': 'application/json' },
+                    body: JSON.stringify(loginData),
+                }
             )
             if (!response.ok) {
                 const error: ApiErrorMessage = await response.json()
                 return handleApiError(error.message)
             }
+            const data: { token: string } = await response.json()
+            //TODO: get a secure cookie instead of local storage late
+            localStorage.setItem('auth_token', data.token)
+            return { success: response.ok, token: data.token }
         } catch (error) {
             const err = error as Error
             return handleApiError(err.message)
diff --git a/src/utils/studentCsvParser.ts b/src/utils/studentCsvParser.ts
index 934876b..ad2e503 100644
--- a/src/utils/studentCsvParser.ts
+++ b/src/utils/studentCsvParser.ts
@@ -21,7 +21,7 @@ export async function makeStudentsFromFile(
                 for (const student of studentsParsed) {
                     students.push({
                         ...student,
-                        id: `id-${Date.now()}-${index}`,
+                        user_id: `id-${Date.now()}-${index}`,
                     })
                     index++
                 }
